docs/core/tutorials/netcore-hosting.md

```diff
@@ -19,40 +19,127 @@ Because hosts are native applications, this tutorial will cover constructing a C
+
+### Step 1 - Find and load CoreCLR
+
+The .NET Core runtime APIs are in *coreclr.dll* (on Windows), in *libcoreclr.so* (on Linux), or in *libcoreclr.dylib* (on macOS). The first step to hosting .NET Core is to load the CoreCLR library. Some hosts probe different paths or use input parameters to find the library while others know to load it from a certain path (next to the host, for example, or from a machine-wide location).
+
+Once found, the library is loaded with `LoadLibraryEx` (on Windows) or `dlopen` (on Linux/Mac).
+
+[!code-cpp[CoreClrHost#1](../../../samples/core/hosting/HostWithCoreClrHost/SampleHost.cpp#1)]
+
+### Step 2 - Get .NET Core hosting functions
+
+CoreClrHost has several important methods useful for hosting .NET Core:
+
+* `coreclr_initialize`: Starts the .NET Core runtime and default AppDomain
+* `coreclr_execute_assembly`: Executes a managed assembly
+* `coreclr_create_delegate`: Creates a function pointer to a managed method
+* `coreclr_shutdown`: Unloads AppDomains and shuts down the .NET Core runtime
+* `coreclr_shutdown_2`: Like `coreclr_shutdown`, but also retrieves the managed code's exit code
+
+After loading the CoreCLR library, the next step is to get references to these functions using `GetProcAddress` (on Windows) or `dlsym` (on Linux/Mac).
+
+[!code-cpp[CoreClrHost#2](../../../samples/core/hosting/HostWithCoreClrHost/SampleHost.cpp#2)]
+
+### Step 3 - Prepare AppDomain properties
+
+Before starting the runtime, we need to prepare properties that will specify AppDomain behavior (especially concerning the assembly loader). The properties are an array of keys and a corresponding array of values that are used when starting the runtime and default AppDomain.
+
+Common AppDomain properties include:
+
+* `TRUSTED_PLATFORM_ASSEMBLIES` This is a list of assembly paths (delimited by ';' on Windows and ':' on Linux) which the AppDomain should prioritize loading and give full trust to (even in partially-trusted domains). This list is meant to contain 'Framework' assemblies and other platform assemblies, similar to the global assembly cache (GAC) in .NET Framework scenarios. Some hosts will put any library next to *coreclr.dll* on this list, others have hard-coded manifests listing trusted assemblies for their purposes.
+* `APP_PATHS` This is a list of paths to probe in for an assembly if it can't be found in the trusted platform assemblies (TPA) list. These paths are meant to be the locations where users' assemblies can be found. Common values for this property include the path the target app was loaded from or other locations where user assets are known to live.
```

Comment:
Ideally we would not list APP_PATHS and APP_NI_PATHS here. Even though this works, it's something we don't want people to use if possible. It's much better to use the TPA list above. For example the default `dotnet.exe` hosting doesn't specify this property by default. The samples here don't make use of this and so there's no reason to include it.

Reply:
Being able to probe for assemblies at runtime could be useful for some hosting scenarios. I think we don't have all of the AppDomain properties documented anywhere so I tried to list the properties most likely to be useful even if this sample doesn't use them.